packages/x-date-pickers/src/internals/hooks/useField/useField.ts

```diff
@@ -470,6 +488,26 @@ export const useField = <
     setSelectedSections: (activeSectionIndex) => setSelectedSections(activeSectionIndex),
   }));
 
+  const handleClearValue = useEventCallback((event: React.MouseEvent, ...args) => {
+    // the click event of the endAdornmnet propagates to the input and triggers the `handleInputClick` handler.
+    event.stopPropagation();
+    event.preventDefault();
+    onClear?.(event, ...(args as []));
+    clearValue();
+    setSelectedSections(0);
+    inputRef?.current?.focus();
+  });
+
+  const handleMouseEnter = useEventCallback((event: React.MouseEvent, ...args) => {
```

Comment:
Have you considered going with css conditions in order to hide/show the icon? 🤔 
Doesn't it cover some use-case?
The CSS solution would be way more lightweight. 🙈 
The current solution also seems to be a bit buggy when using tabulation to focus between elements. 🤔 

https://github.com/mui/mui-x/assets/4941090/31b66dc9-6c27-4809-9756-984a52383109

Reply:
I would personally prefer a css solution myself 🙈 It would be helpful to be able to play around with the `visibility`

The reason I chose this implementation is to be able to control the same behavior from a shared piece of code 🤔 

A css solution would look something like: 

```javascript
  '&:focus .clearButton': {
    visibility: 'visible',
  },
  '&:hover .clearButton': {
    visibility: 'visible',
  },
```

But we would need to do this from the parent, and in our case the parent could be `DateField`, `DateTimeField`, `TimeField`, etc... soo

option 1: we do this for every field :(
option 2: we find a nicer, more reusable way to do this - preferable but **VERY** open to suggestions 😅 
An idea I had for this is was to have a css class that we pass to the parent from the new hook, but seems kind of shady to me. Or to have a wrapper for the parent and do this from there 🤔

Reply:
I'm not sure that there is a best solution for this case, but 2 options come to my mind:
1. change the used `textField` slot component in `useDesktopPicker` for a `styled` TextField having those mentioned styles
2. create a wrapper component (like you said) used on top of `Field` or `TextField` and add those styles there.

1st option sounds and looks cleaner also should result in 0 DOM changes, but it would not handle cases, where a custom slot is used.
2nd option would alter the DOM, which might be a BC in certain niche cases for someone and is not as clean in general, but it should also handle cases (theoretically), when a custom `textIField` slot is used. 🤔 

What do you think? @noraleonte @flaviendelangle

Reply:
For solution 1., how would we allow custom `textfield` to support this button?
They would have to copy paste this CSS ? 
If so, that's clearly not great, but maybe that's the better we can do :/

Reply:
@LukasTy How do you feel about about processing the `fieldProps` inside the `useClearEndAdornment` somewhat like this:
```javascript
  const fieldProps = {
    ...forwardedFieldProps,
    sx: {
      '&:focus .clearButton': {
        visibility: 'visible',
      },
      '&:hover .clearButton': {
        visibility: 'visible',
      },
      ...forwardedFieldProps?.sx,
    },
  };

  return { InputProps, fieldProps };
```
as @flaviendelangle said, this would take for granted that the child component supports the `sx`  prop.

Reply:
I'm not sure we need to strive for providing a solution covering all bases on the get go.
After all, look at [`Autocomplete` doing the same](https://mui.com/material-ui/react-autocomplete/#custom-input), if you are not using `TextField`—no clear icon for you. 🙈

Reply:
I think the clearing icon should be at the field level, not at the picker level. That way standalone fields do have it.

For me the priority is (in descending order):

1. Do not replicate the CSS for each of our fields
2. Make it simple enough for custom fields with similar layout (Joy being the most important IMHO) to have this feature
3. Make it simple enough for other custom fields to have this feature (clearly that one is super super optional)

With that in mind, I think having it in `useClearEndAdornment` (which could be renamed if needed) would solve 1. and I think 2.